src/backend_common.cc

```diff
@@ -1372,4 +1372,64 @@ GetRequestId(TRITONBACKEND_Request* request)
   return std::string("[request id: ") + request_id + "] ";
 }
 
+TRITONSERVER_Error*
+ValidateStringBuffer(
+    const char* buffer, size_t buffer_byte_size,
+    const size_t expected_element_cnt, const char* input_name,
+    std::vector<std::pair<const char*, const uint32_t>>* str_list)
```

Comment:
Do we need to document that the lifetime of `str_list` is dependent on the validity of `buffer` object?